parse/parse.go

```diff
@@ -52,22 +51,46 @@ func GoListAgnostic(stdIn io.Reader) (deps types.ProjectList, err error) {
 	decoder := json.NewDecoder(strings.NewReader(string(johnnyFiveNeedInput)))
 
 	for {
-		var mod types.GoListModule
+		var mod map[string]interface{}
 
-		err = decoder.Decode(&mod)
-		if err == io.EOF {
-			err = nil
+		decodeErr := decoder.Decode(&mod)
+
+		if decodeErr == io.EOF {
 			break
 		}
-		if err != nil {
+		if decodeErr != nil {
+			err = decodeErr
 			break
 		}
 
-		project, err := modToProjectList(mod)
-		if _, ok := err.(*NoVersionError); ok {
+		if module, ok := mod["Module"].(map[string]interface{}); ok {
```

Comment:
The easiest way I found to kind of tell if something was of `go list -m all` style versus `go list -deps -json` was this. I'm doing type checks on map keys, basically looking for existence. This lost us some of the nicer json parsing into a struct, but the amount of data we end up using is so minimal this didn't seem too wild. If a `Module` key exists, I know I'm in `go list -deps` town. If it doesn't, I look for `Path`, which is `go list -m all` town. If we find none of that, we just parse like it's a regular list (which we should drop in the future, there's not much of a reason to keep it around).

Reply:
Neato. The explanation above makes total sense now, while it's on my mind. It feels like adding some code comments around the "why" of these check might avoid head scratching down the road.

BTW, this looks very nice to me: `207` vs `49`

```
$ go list -json -m all | ./nancy sleuth
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Summary                       ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━┫
┃ Audited Dependencies    ┃ 207 ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━┫
┃ Vulnerable Dependencies ┃ 0   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━┛
(base) MBP-DRollo5:nancy bhamail$ go list -json -deps | ./nancy sleuth 
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Summary                      ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━┫
┃ Audited Dependencies    ┃ 49 ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━┫
┃ Vulnerable Dependencies ┃ 0  ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━┛
```